src/main/java/org/opendatakit/aggregate/query/submission/QueryByUIFilterGroup.java

```diff
@@ -135,50 +135,72 @@ public QueryByUIFilterGroup(IForm form, FilterGroup filterGroup, CompletionFlag
         FormElementModel fem = FormElementModel.retrieveFormElementModel(form, decodeKey);
         FilterOperation op = UITrans.convertFilterOperation(rf.getOperation(), rf.getVisibility());
 
-        String value = rf.getInput();
-        Object compareValue = null;
+        Object compareValue = getCompareValue(fem, rf.getInput());
+
         switch (fem.getElementType()) {
           case BOOLEAN:
-            compareValue = WebUtils.parseBoolean(value);
+            super.addFilter(fem, op, compareValue);
             break;
           case JRDATETIME:
-            compareValue = WebUtils.parseDate(value);
-            super.addFilterChildren(fem, column.getChildColumnCode(), op, compareValue);
-            continue;
+            if (fem.getFormDataModel() == null && fem.getElementName().contains("meta-"))
+              super.addFilter(fem, op, compareValue);
+            else if (fem.getFormDataModel() != null)
+              super.addFilterChildren(fem, column.getChildColumnCode(), op, compareValue);
+            else
+              throw new IllegalArgumentException("Unrecognized dateTime field");
+            break;
           case JRDATE:
-            compareValue = WebUtils.parseDate(value);
             super.addFilterChildren(fem, column.getChildColumnCode(), op, compareValue);
-            continue;
+            break;
           case JRTIME:
-            compareValue = WebUtils.parseDate(value);
             super.addFilterChildren(fem, column.getChildColumnCode(), op, compareValue);
-            continue;
+            break;
           case INTEGER:
-            compareValue = Long.valueOf(value);
+            super.addFilter(fem, op, compareValue);
```

Comment:
How about grouping together these four cases so this only appears once:
`super.addFilter(fem, op, compareValue);`